lib/helper/Nightmare.js

```diff
@@ -156,7 +157,30 @@ class Nightmare extends Helper {
     });
   }
 
+  _beforeSuite() {
+    if (!this.options.restart)
+      this._startBrowser();
+  }
+
   _before() {
+    if (this.options.restart)
+        this._startBrowser();
+    return this.browser;
+  }
+
+  _after() {
+    if (this.options.restart)
+        return this._stopBrowser();
+    else
+        return this.browser.cookies.clearAll();
```

Comment:
Clear the cookies before handing it over to the next test. Should this just return the browser without clearing it? I don't know.